data-prepper-plugins/grok-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/grok/GrokProcessorConfig.java

```diff
@@ -61,7 +61,7 @@ public class GrokProcessorConfig {
     @JsonPropertyDescription("Specifies a parent-level key used to store all captures. Default value is <code>null</code> which will write captures into the root of the event.")
     private String targetKey = DEFAULT_TARGET_KEY;
 
-    @JsonProperty(BREAK_ON_MATCH)
+    @JsonProperty(value = BREAK_ON_MATCH, defaultValue = true)
```

Comment:
The `defaultValue` configuration on `@JsonValue` only accepts string values.
```suggestion
    @JsonProperty(value = BREAK_ON_MATCH, defaultValue = "true")
```